pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1641,6 +1650,45 @@ func (l *LibvirtDomainManager) ListAllDomains() ([]*api.Domain, error) {
 	return list, nil
 }
 
+func (l *LibvirtDomainManager) ListAllDomainsWithRuntimeInfo() ([]*api.Domain, error) {
+
+	doms, err := l.virConn.ListAllDomains(libvirt.CONNECT_LIST_DOMAINS_ACTIVE | libvirt.CONNECT_LIST_DOMAINS_INACTIVE)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []*api.Domain
+	for _, dom := range doms {
+		domain, err := util.NewDomain(dom)
+		if err != nil {
+			if domainerrors.IsNotFound(err) {
+				continue
+			}
+			return list, err
+		}
+		spec, err := l.getDomainSpecWithRuntimeInfo(dom)
```

Comment:
this function appears to be nearly 95% the same as ListAllDomains, The only difference is this single function call. Can centralized a helper function be created here that toggles how the spec is retrieved based on a boolean?

Usually I wouldn't care much, but this area is so convoluted i'd hate for us to make a fix to one of these two functions in the future and not realize the same change applies to both cases.

Reply:
This is a very reasonable point! I'll add a changeset to merge those